dbms/src/DataStreams/ParallelInputsProcessor.h

```diff
@@ -205,27 +217,53 @@ class ParallelInputsProcessor
 
         try
         {
-            while (!finish)
-            {
-                InputData unprepared_input;
-                {
-                    std::lock_guard lock(unprepared_inputs_mutex);
+            loop(thread_num);
+        }
+        catch (...)
+        {
+            exception = std::current_exception();
```

Comment:
how about using `getCurrentExceptionMessage(true)`

Reply:
The `exception` is a `std::exception_ptr` not string.